src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -582,27 +589,100 @@ private CodeRefactoring FilterOnUIThread(CodeRefactoring refactoring, Workspace
 
-                    foreach (var group in actions.GroupBy(a => a.Priority))
+                    var nonSuppressionActions = actions.Where(a => !IsTopLevelSuppressionAction(a.CodeAction));
+                    AddSuggestedActionsSet(nonSuppressionActions, diag, nonSuppressionSets);
+
+                    var suppressionActions = actions.Where(a => IsTopLevelSuppressionAction(a.CodeAction));
+                    AddSuggestedActionsSet(suppressionActions, diag, suppressionSets);
+                }
+
+                var sets = nonSuppressionSets.ToImmutableAndFree();
+
+                if (suppressionSets.Count > 0)
+                {
+                    // Wrap the suppression/configuration actions within another top level suggested action
+                    // to avoid clutter in the light bulb menu.
+                    var wrappingSuggestedAction = new SuggestedActionWithNestedActions(
+                        ThreadingContext, _owner, workspace, _subjectBuffer, this,
+                        codeAction: new SolutionChangeAction(EditorFeaturesWpfResources.Configure_or_Suppress_issues, createChangedSolution: _ => null),
+                        nestedActionSets: suppressionSets.ToImmutable());
+
+                    var (span, category) = CombineSpansAndCategory(suppressionSets);
+                    var wrappingSet = new SuggestedActionSet(
+                        category,
+                        actions: SpecializedCollections.SingletonEnumerable(wrappingSuggestedAction),
+                        title: EditorFeaturesWpfResources.Configure_or_Suppress_issues,
+                        priority: SuggestedActionSetPriority.None,
+                        applicableToSpan: span);
+                    sets = sets.Add(wrappingSet);
+                }
+
+                suppressionSets.Free();
+                return sets;
+            }
+
+            private static void AddSuggestedActionsSet(
+                IEnumerable<SuggestedAction> actions,
+                CodeFixGroupKey diag,
+                ArrayBuilder<SuggestedActionSet> sets)
+            {
+                foreach (var group in actions.GroupBy(a => a.Priority))
+                {
+                    var priority = GetSuggestedActionSetPriority(group.Key);
+
+                    // diagnostic from things like build shouldn't reach here since we don't support LB for those diagnostics
+                    Debug.Assert(diag.Item1.HasTextSpan);
+                    var category = GetFixCategory(diag.Item1.Severity);
+                    sets.Add(new SuggestedActionSet(category, group, priority: priority, applicableToSpan: diag.Item1.TextSpan.ToSpan()));
+                }
+            }
+
+            private static (Span? span, string category) CombineSpansAndCategory(IEnumerable<SuggestedActionSet> sets)
+            {
+                int minStart = -1;
+                int maxEnd = -1;
+                string category = PredefinedSuggestedActionCategoryNames.CodeFix;
```

Comment:
can we add some comments explaining what's going on.

Reply:
Added comments.